### PR TITLE
Fix map attribute name

### PR DESCRIPTION
**Pull request by cbumstead**
### Why did you make these changes?

To support using attribute name when using a map attribute

### What's changed in these changes?

Modified map.ts, added a unit test

### What do you especially want to get reviewed?

map.ts changes

### Submission Type

* [x] Bugfix

### All Submissions

* [x] Have you added an explanation of what your changes?
* [x] Have you written new tests for your changes, as applicable?
* [ ] Have you checked potential side effects that could make bad impacts to other services?

### CODE REVIEW

**Comment by benhutchins** [comment]
I appreciate the pull request @cbumstead!

I'll need to revise your edits to `package.json` before merging, but we can definitely get this merged in.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "dyngoose",
-  "version": "3.1.3",
+  "name": "@hixme/dyngoose",
```
Comment (benhutchins) [inline package.json:2]: Don't want to rename the package here.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+  "version": "3.2.0",
```
Comment (benhutchins) [inline package.json:3]: Do not need to manually bump the version for release, this is done automatically from the CI release process.

**Review by benhutchins — CHANGES_REQUESTED** [review]

**Comment by cbumstead** [comment]
thanks,  I revised the package.json

**Comment by benhutchins** [comment]
@cbumstead Thank you so much for keeping this PR alive with your recent update. I was reviewing it to get it merged in noticed the implementation was changing the way the map attribute's `attributes` object mapped from attribute names to property names. I pulled your change in and revised it to keep it consistent with the way the table attributes work, but this issue has been fixed in the new release 4.1.0.

Also, love the vscode launch config addition. Makes sense to help others who use vscode and the integrated debugger functionality.

Thank you for the contribution!

**Pull request closed without merging**